File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let xy=yellow+brown;
    let y=0;
    for(let x=2;x<~~Math.sqrt(xy)+1;x++){
        if(xy%x===0){
            y=xy/x;
            if(x+y===brown/2+2)
                return [Math.max(x,y),Math.min(x,y)];
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    xy=brown+yellow
    y=0
    for x in range(2,int(xy**0.5)+1):
        if xy%x==0:
            y=xy/x
        if x+y==brown/2+2:
            return [max(x,y),min(x,y)]
